lib/export/dwca/data.rb

```diff
@@ -629,37 +887,664 @@ def biological_associations_resource_relationship_tmp
       return nil if biological_associations_extension.nil?
       @biological_associations_resource_relationship_tmp = Tempfile.new('biological_resource_relationship.xml')
 
-      content = nil
-
       if no_records?
-        content = "\n"
+        @biological_associations_resource_relationship_tmp.write("\n")
       else
-        content = Export::CSV::Dwc::Extension::BiologicalAssociations.csv(biological_associations_extension, biological_association_relations_to_core)
+        Export::CSV::Dwc::Extension::BiologicalAssociations.csv(
+          biological_associations_extension,
+          biological_association_relations_to_core,
+          output_file: @biological_associations_resource_relationship_tmp
+        )
       end
 
-      @biological_associations_resource_relationship_tmp.write(content)
       @biological_associations_resource_relationship_tmp.flush
       @biological_associations_resource_relationship_tmp.rewind
       @biological_associations_resource_relationship_tmp
     end
 
     def media_tmp
-      return nil if media_extension.nil? || media_extension.empty?
+      return nil if @media_extension.nil?
       @media_tmp = Tempfile.new('media.xml')
 
-      content = nil
       if no_records?
-        content = "\n"
+        @media_tmp.write("\n")
       else
-        content = Export::CSV::Dwc::Extension::Media.csv(media_extension[:collection_objects], media_extension[:field_occurrences])
+        # Write directly to @media_tmp to avoid loading entire dataset into memory
+        media_extension_optimized(@media_tmp)
       end
 
-      @media_tmp.write(content)
       @media_tmp.flush
       @media_tmp.rewind
       @media_tmp
     end
 
+    # SQL fragment: Creative Commons license mapping
+    def creative_commons_license_case_sql
```

Comment:
To be safer we need a CONSTANT and string concat referencing that constant here.